src/fiber/channel.rs

```diff
@@ -2683,6 +2730,34 @@ impl PendingTlcs {
             .cloned()
             .collect()
     }
+
+    pub fn get_oldest_failed_tlcs(&self) -> Vec<TLCId> {
+        let mut failed_tlcs = self
+            .tlcs
+            .iter()
+            .filter(|tlc| {
+                matches!(tlc.removed_reason, Some(RemoveTlcReason::RemoveTlcFail(_)))
+                    && matches!(
+                        tlc.status,
+                        TlcStatus::Inbound(InboundTlcStatus::RemoveAckConfirmed)
+                            | TlcStatus::Outbound(OutboundTlcStatus::RemoveAckConfirmed)
+                            | TlcStatus::Outbound(OutboundTlcStatus::RemoveWaitAck)
+                    )
+            })
+            .map(|tlc| (tlc.tlc_id, tlc.removed_confirmed_at.unwrap_or(u64::MAX)))
+            .collect::<Vec<_>>();
+
+        let keep_failed_num = 2;
+        if failed_tlcs.len() >= keep_failed_num {
```

Comment:
I think it should be `failed_tlcs.len() > keep_failed_num`

Reply:
`failed_tlcs.len() > keep_failed_num` means we keep 2 confirmed failed tlcs, the minimal safe way is we only keep another extra confirmed failed tlc.

I think both are OK, I changed as you suggested which may sound safer.